libp2p-xtra/examples/hello_world_dialer.rs

```diff
@@ -37,11 +39,22 @@ async fn main() -> Result<()> {
         .create(None)
         .spawn_global();
 
-    endpoint_addr
-        .send(Connect(opts.multiaddr.clone()))
-        .await
-        .unwrap()
-        .unwrap();
+    let dialer_constructor = {
+        // XXX: Can this be done nicer???
+        let connect_addr = opts.multiaddr.clone();
+        let endpoint_addr = endpoint_addr.clone();
+        move || {
+            let connect_addr = connect_addr.clone();
+            let endpoint_addr = endpoint_addr.clone();
+            dialer::Actor::new(endpoint_addr, connect_addr)
+        }
+    };
```

Comment:
> Can this be done nicer?

I don't think so. I would inline the second set of clones though, but it's just style.

Reply:
happy to inline the second set of clones, makes it *a bit* nicer. thanks for looking at this bit :)